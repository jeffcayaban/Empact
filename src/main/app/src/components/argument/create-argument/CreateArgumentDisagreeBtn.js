import React from "react";
import { Button } from "react-bootstrap";
import { Route } from "react-router-dom";

/**
 * This button is used for places when an argument is disagreed on.
 */

class CreateArgumentDisagreeBtn extends React.PureComponent {

    render() {
        const { isContentCreator, closeHandler, contentId, contentType } = this.props;

        if (isContentCreator) {
            return <Button onClick={() => closeHandler()}>No</Button>;
        } else {
            return (
                <Route render={({ history }) => (
                    <Button onClick={() => { history.push(`/${contentType.toLowerCase()}/${contentId}/disagree`); }}>
                        Disagree
                    </Button>
                )}/>
            )
        }
    }
}

export default CreateArgumentDisagreeBtn;