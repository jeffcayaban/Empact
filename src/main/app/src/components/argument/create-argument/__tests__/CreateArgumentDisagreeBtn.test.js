import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import CreateArgumentDisagreeBtn from "../CreateArgumentDisagreeBtn";

describe('CreateArgumentDisagreeBtn', () => {

    it('should render successfully given the user is the content creator', () => {
        const component = shallow(
            <CreateArgumentDisagreeBtn
                isContentCreator={true}
                closeHandler={jest.fn()}
                contentId={"testId"}
                contentType={"testContentType"}
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully given the user is not the content creator', () => {
        const component = shallow(
            <CreateArgumentDisagreeBtn
                isContentCreator={false}
                closeHandler={jest.fn()}
                contentId={"testId"}
                contentType={"testContentType"}
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
