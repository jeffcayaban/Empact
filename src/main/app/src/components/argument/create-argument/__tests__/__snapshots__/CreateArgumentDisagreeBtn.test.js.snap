// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateArgumentDisagreeBtn should render successfully given the user is not the content creator 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`CreateArgumentDisagreeBtn should render successfully given the user is the content creator 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  onClick={[Function]}
>
  No
</Button>
`;
